### PR TITLE
Implement kioskOriginLocationId prop

### PR DESCRIPTION
**Pull request by andreeaceachir142**
# What 

- Implement kioskOriginLocationId prop 
- Center the sidebar instead of having it placed in the left side
- Use the `fitBounds` function when having both `kioskOriginLocationId` and `locationId` props
- Restrict the kiosk to only be on desktop devices
- Create a new function `fitBoundsLocation` which is reused in three different places

# How 

- Refactor the `onLocationIdChanged` function to happen in a `useEffect`, and make sure the map is ready at the time the changes are happening 
- Handle both the `kioskOriginLocationId` and the `locationId` in the same `useEffect`
- Create helper functions for getting the desktop padding to the left and bottom, and the mobile padding to the bottom
- Do not allow location selection or clicking the location if it's the `kioskOriginLocationId` (meaning that it should not take the selection pin, and it should also not be selected to show the Location Details page)
- Make the kiosk icon double the size based on the display rule (UX Spec)
- Create a new function `fitBoundsLocation` which calculates the bbox of the location, takes a MapsIndoors instance and padding left and bottom, then it calls `fitBounds` method.

### CODE REVIEW

File: packages/map-template/src/components/MapTemplate/MapTemplate.jsx [hunk 1]
```diff
@@ -320,12 +339,13 @@ function MapTemplate({ apiKey, gmApiKey, mapboxAccessToken, venue, locationId, p
 
     /**
      * Handle the clicked location on the map.
-     * Set the current location if not in directions mode.
+     * Set the current location if not in directions mode,
+     * and if the clicked location is not the same as the kioskOriginLocationId.
```
Comment (ammapspeople) [inline packages/map-template/src/components/MapTemplate/MapTemplate.jsx:343]: Should we motivate/describe why?

File: packages/map-template/src/components/Sidebar/Modal/Modal.jsx [hunk 2]
```diff
@@ -41,7 +44,15 @@ function Modal({ children, isOpen }) {
         }
     }, [contentRef]);
 
-    return <div ref={modalRef} className={`modal ${isOpen ? 'modal--open' : ''} ${fullHeight ? 'modal--full' : ''} ${substeps ? 'modal--substeps' : ''}`}>
+    return <div ref={modalRef}
+        className={`modal ${isOpen ? 'modal--open' : ''} ${fullHeight ? 'modal--full' : ''} ${substeps ? 'modal--substeps' : ''}`}
+        style={{
```
Comment (ammapspeople) [inline packages/map-template/src/components/Sidebar/Modal/Modal.jsx:49]: Could we add a BEM modifier instead?

**Review by ammapspeople — COMMENTED** [review]
Do we need something added to READMEs and other documentation here and there?

File: packages/map-template/src/components/Map/Map.jsx [hunk 3]
```diff
@@ -264,6 +244,46 @@ function Map({ onLocationClick, onVenueChangedOnMap }) {
...
 
+    /*
+    * React on changes in the kioskOriginLocationId and locationId.
+    */
+    useEffect(() => {
+        if (isMapReady && mapsIndoorsInstance) {
+            if (kioskOriginLocationId && isDesktop) {
+                window.mapsindoors.services.LocationsService.getLocation(kioskOriginLocationId).then(kioskOriginLocation => {
+                    if (kioskOriginLocation) {
+                        // Replace icon with "you are here" icon
```
[the remark this answers is not in the document]
Comment (andreeaceachir142) [reply packages/map-template/src/components/Map/Map.jsx]: Updated comments. ✅

File: packages/map-template/src/helpers/fitBoundsLocation.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { calculateBounds } from "./CalculateBounds";
+
+/**
```
[the remark this answers is not in the document]
Comment (andreeaceachir142) [reply packages/map-template/src/helpers/fitBoundsLocation.js:3]: Described params. ✅

File: packages/map-template/src/components/Map/Map.jsx [hunk 5]
```diff
@@ -264,6 +244,46 @@ function Map({ onLocationClick, onVenueChangedOnMap }) {
...
+                    }
+                });
+            } else if (locationId) {
+                window.mapsindoors.services.LocationsService.getLocation(locationId).then(location => {
+                    if (location) {
+                        // Set the floor to the one that the location belongs to.
+                        const locationFloor = location.properties.floor;
+                        mapsIndoorsInstance.setFloor(locationFloor);
+
+                        fitBoundsLocation(location, mapsIndoorsInstance, 0, getDesktopPaddingLeft());
```
[the remark this answers is not in the document]
Comment (andreeaceachir142) [reply packages/map-template/src/components/Map/Map.jsx]: Good point, implemented functionality for handling mobile as well. ✅

File: packages/map-template/src/components/MapTemplate/MapTemplate.jsx (hunk 1 above)
Comment (andreeaceachir142) [reply packages/map-template/src/components/MapTemplate/MapTemplate.jsx]: Added a more detailed description. ✅

**Comment by andreeaceachir142** [comment]
@ammapspeople Regarding this comment https://github.com/MapsPeople/web-ui/pull/281#pullrequestreview-1733840231, I was thinking to update the documentation everywhere at once, ideally at the end when we have an overview of all the combinations of parameters and so on. I could also add it now, for the `kioskOriginLocationId` only. Do you think that would be a better approach?

File: packages/map-template/src/components/Sidebar/Modal/Modal.jsx (hunk 2 above)
Comment (andreeaceachir142) [reply packages/map-template/src/components/Sidebar/Modal/Modal.jsx]: That's a good idea, I added a BEM modifier instead.

File: packages/map-template/src/components/MapTemplate/MapTemplate.jsx (hunk 1 above)
Comment (ammapspeople) [reply packages/map-template/src/components/MapTemplate/MapTemplate.jsx]: 👏

File: packages/map-template/src/components/Sidebar/Modal/Modal.jsx (hunk 2 above)
Comment (ammapspeople) [reply packages/map-template/src/components/Sidebar/Modal/Modal.jsx]: 🙌

**Comment by andreeaceachir142** [comment]
Ready to be reviewed again @ammapspeople

**Review by ammapspeople — APPROVED** [review]
👏

**Pull request merged**